### DISCUSSION TITLE
Is there a way I can prevent NSFW images from coming in results?

### DISCUSSION THREAD

**Discussion by TechPro424** [arifszn/reddit-image-fetcher#6] [Q&A] [answered]
Is there a way I can prevent NSFW images from coming in results? I am trying to make a family friendly Discord bot and I am using this package for meme functionality, and I do not want NSFW memes showing up. Also, is there a way I can search for a particular image, say `minecraft` for a Minecraft meme.

**Comment by arifszn** [comment, marked as answer]
* You have two options to prevent NSFW images.
1. The response has NSFW flag. You can do a filter based on that flag after getting the response.
![image](https://user-images.githubusercontent.com/45073703/131542635-47dd22cf-7f8d-4d20-82de-514f95365c3a.png)

2. You can specify your own non-NSFW subreddits library.

I will try to add an option for filtering out NSFW images in the next version if it is needed.

<br/> <br/>

* To get minecraft memes only, you would have to specify custom subreddits library. Example:

```
RedditImageFetcher.fetch({
    type: 'custom',
    total: 1, 
    subreddit: ['MinecraftMemes']
}).then(result => {
    console.log(result);
});
```

This would only return images of minecraft memes.

**Reply by TechPro424** [reply]
Thanks! A filter for NSFW images in the next version, as well as the functionality to search the whole of Reddit for images related to a search term, would be welocme.
